### PR TITLE
Xi-Plus proposal

### PR DESCRIPTION
**Pull request by Xi-Plus**
---
A good PR should include the following parts:

- [x] A clear title
- [x] A meaningful message for PR, as well as its commits
- [x] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [x] excluding any irrelevant files, such as binaries, text files, or dot files
- [x] passing tests/CI

### CODE REVIEW

File: proposal/Xi-Plus/README.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
+==================================
+Proposal for shared-data-encrypter
+==================================
+
+Basic information
+=================
+
+GitHub repository: https://github.com/Xi-Plus/shared-data-encrypter
+
+An encryption library provides encrypted data that you want to send to multiple people. You don't need to send lots of different copies. And the recipients don't share the key.
```
Comment (yungyuc) [inline proposal/Xi-Plus/README.rst:10]: I am not sure cryptography involves real numbers.  If your applications don't, please make a good argument why it should use the techniques covered in the course.

File: proposal/Xi-Plus/README.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+Write comments and follow style guide. Use some tools like autodoc to generate docs automatically.
+
+Write unit tests to check function works.
+
+Schedule
+========
+
+Gantt chart:
+
+.. image:: gantt-chart.png
```
Comment (yungyuc) [inline proposal/Xi-Plus/README.rst:73]: It looks nice.  Was it made by notion or something else?

File: proposal/Xi-Plus/README.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+   - It means that user key is asymmetric.
+- Reduce calculation when users changed their password.
+   - Decrypt "encrypted user private key", then encrypt the key with new password. No need to change encrypted data key for the user.
+- Reduce calculation when data is modified.
+   - Encrypt the data with data public key only. No need to change anything else.
+
+API description
+===============
+
+- class Data
```
Comment (yungyuc) [inline proposal/Xi-Plus/README.rst:51]: Looks nice.

**Review by yungyuc — COMMENTED** [review]
LGTM.  I will merge once I grade for all the proposals.

File: proposal/Xi-Plus/README.rst (hunk 1 above)
Comment (Xi-Plus) [reply proposal/Xi-Plus/README.rst:10]: Cryptography algorithm usually invokes integers (e.g., RSA, AES). Some algorithm use matrix calculation in encryption processes.

File: proposal/Xi-Plus/README.rst (hunk 2 above)
Comment (Xi-Plus) [reply proposal/Xi-Plus/README.rst:73]: Google sheets with [ProjectSheet planning](https://workspace.google.com/marketplace/app/projectsheet_planning/1080532206233) add-ons.

**Pull request merged**